pkg/entrypoint/entrypointer.go

```diff
@@ -153,9 +173,26 @@ func (e Entrypointer) Go() error {
 		}
 	}
 
-	if err != nil && e.BreakpointOnFailure {
+	var ee *exec.ExitError
+	switch {
+	case err != nil && e.BreakpointOnFailure:
 		logger.Info("Skipping writing to PostFile")
-	} else {
+	case e.OnError == ContinueOnError && errors.As(err, &ee):
+		// with continue on error and an ExitError, write non-zero exit code and a post file
+		exitCode := strconv.Itoa(ee.ExitCode())
+		output = append(output, v1beta1.PipelineResourceResult{
+			Key:        "ExitCode",
+			Value:      exitCode,
+			ResultType: v1beta1.InternalTektonResultType,
+		})
+		e.WritePostFile(e.PostFile, nil)
+		e.WriteExitCodeFile(e.StepPath, exitCode)
+	case err == nil:
+		// if err is nil, write zero exit code and a post file
+		e.WritePostFile(e.PostFile, nil)
+		e.WriteExitCodeFile(e.StepPathLink, "0")
+	default:
+		// for a step without continue on error and any error, write a post file with .err
```

Comment:
creating an `InternalTektonResultType` with a failed exit code similar to `StartedAt` time which is then read while updating the pod status.